opent2t.p.light/com.wink.lightbulb/js/thingTranslator.js

```diff
@@ -0,0 +1,243 @@
+    var dim = {
+         "href": "/dim",
+         "rt": ["oic.r.light.dimming"],
+         "if": ["oic.if.a", "oic.if.baseline"]
+    }
+
+    // Include the values is expand is specified
+    if (expand) {
+        power.id = 'power';
+        power.value = stateReader.get('powered');
+
+        dim.id = 'dim';
+        dim.dimmingSetting = scaleDeviceBrightnessToTranslatorBrightness(stateReader.get('brightness'));
+        dim.range = [0,100];
+    }
+
+    return {
+        opent2t: {
+            schema: 'opent2t.p.light',
+            translator: 'opent2t-translator-com-wink-lightbulb',
+            control_id: providerSchema['object_id']
+        },
+        pi: providerSchema['uuid'],
+        mnmn: providerSchema['device_manufacturer'],
+        mnmo: providerSchema['manufacturer_device_model'],
+        n: providerSchema['name'],
+        rt: ['opent2t.p.light'],
+        entities: [
+            {
+                rt: ['opent2t.d.light'],
+                di: 'F8CFB903-58BB-4753-97E0-72BD7DBC7933',
+                icv: 'core.1.1.0',
+                dmv: 'res.1.1.0',
+                resources: [
+                    power,
+                    dim
+                ]
+            }
+        ]
+    };
+}
+
+var deviceId;
+var deviceType = 'light_bulbs';
+var winkHub;
+
+// This translator class implements the 'org.opent2t.sample.lamp.superpopular' interface.
+class Translator {
+
+    constructor(deviceInfo) {
+        console.log('Wink Lightbulb initializing...');
+
+        validateArgumentType(deviceInfo, "deviceInfo", "object");
+        
+        deviceId = deviceInfo.deviceInfo.id;
+        winkHub = deviceInfo.hub;
+
+        console.log('Wink Lightbulb initializing...Done');
+    }
+
+    // exports for the entire schema object
+
+    // Queries the entire state of the lamp
+    // and returns an object that maps to the json schema org.opent2t.sample.lamp.superpopular
+    get(expand) {
+
+        console.log('Getting from Wink');
+
+        return winkHub.getDeviceDetailsAsync(deviceType, deviceId)
+            .then((response) => {
+                return providerSchemaToPlatformSchema(response.data, expand);
+            });
+    }
+
+    getDeviceResource(di, resourceId) {
+        return this.get(true)
+            .then(response => {
+                return findResource(response, di, resourceId);
+        });
+    }
+
+    /**
+     * Updates the specified resource with the provided payload.
+     */
+    postDeviceResource(di, resourceId, payload) {
+        var putPayload = resourceSchemaToProviderSchema(resourceId, payload);
+
+        return winkHub.putDeviceDetailsAsync(deviceType, deviceId, putPayload)
+            .then((response) => {
+                var schema = providerSchemaToPlatformSchema(response.data, true);
+
+                return findResource(schema, di, resourceId);
+            });
+    }
+
+    getLDevicesPower(deviceId) {
+        return this.getDeviceResource(deviceId, "oic.r.switch.binary");
```

Comment:
For this, and other uses of get/postDeviceResource, you should be looking up resources by their id, not their resource type, so for this call it would be 'power', not 'oic.r.switch.binary'.

For this Light platform, there are no duplicated resource types, so it would be fine in theory to look up resources by their type, but that wouldn't work for other platforms, like a 'thermostat' that has multiple 'oic.r.temperature' resources.